packages/urlstate/utils.ts

```
@@ -35,6 +35,23 @@ export const typeOf = (val: unknown): Type => {
+  return type !== "object" && type !== "array";
+};
+
+export type Primitive =
+  | string
+  | Date
+  | boolean
+  | number
+  | null
+  | undefined
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | Function
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)